### PR TITLE
build: minor fix for 32-bit archs

### PR DESCRIPTION
**Pull request by goaway**
Commit Message: build: minor fix for 32-bit archs
Additional Description: Fixes an implicit cast error building for 32-bit Android here.
Risk Level: Low
Testing: Builds locally

Signed-off-by: Mike Schore <mike.schore@gmail.com>

### CODE REVIEW

File: source/common/stats/symbol_table_impl.h [hunk 1]
```diff
@@ -466,7 +466,8 @@ class StatName {
    * hasher and comparator.
    */
   absl::string_view dataAsStringView() const {
-    return {reinterpret_cast<const char*>(data()), dataSize()};
+    return {reinterpret_cast<const char*>(data()),
+            static_cast<absl::string_view::size_type>(dataSize())};
```
Comment (wrowe) [inline source/common/stats/symbol_table_impl.h:470]: Isn't the underlying problem the abuse of uint64_t in line 409 above for dataSize()?

There's a significant penalty for using a 64 bit int in 32 bit contexts.

Comment (lizan) [reply source/common/stats/symbol_table_impl.h:470]: @jmarantz ?

Comment (goaway) [reply source/common/stats/symbol_table_impl.h:470]: I don't mind changing the underlying type of `dataSize`, but I don't really have any context on the original choice.

Comment (jmarantz) [reply source/common/stats/symbol_table_impl.h:470]: Personally I'd prefer size_t for this; IIUC Envoy style has veered toward using uint64_t. @mattklein123 may have more context.

I'm fine with either:
 * this change, which makes sense in the context of returning a string_view
 * changing dataSize() et al to return size_t, which makes sense generally IMO

I slightly prefer the latter but I'm OK with this. @mattklein123 what do you prefer?

Comment (mattklein123) [reply source/common/stats/symbol_table_impl.h:470]: Sure we can use size_t. Typically we have used explicit types for things that can wrap, but for something like this size_t seems fine.

/wait

Comment (goaway) [reply source/common/stats/symbol_table_impl.h:470]: `uint64_t` seems to be used extensively here to represent a number of bytes. `dataSize()` is also called in a number of locations. Just to clarify, is the proposal to switch `size_t` for all of these cases?

Comment (mattklein123) [reply source/common/stats/symbol_table_impl.h:470]: Yes size_t seems appropriate for number of bytes.

Comment (goaway) [reply source/common/stats/symbol_table_impl.h:470]: Despite updating the type used for representing byte counts throughout the file, I left this cast in; I don't know what sort of contract `absl::string_view::size_type` has with respect to `size_t`.

If folks feel it's redundant though now, I can remove it.

File: source/common/stats/symbol_table_impl.cc [hunk 2]
```diff
@@ -451,8 +451,8 @@ StatNameStorage::StatNameStorage(absl::string_view name, SymbolTable& table)
     : StatNameStorageBase(table.encode(name)) {}
 
 StatNameStorage::StatNameStorage(StatName src, SymbolTable& table) {
-  const uint64_t size = src.size();
-  MemBlockBuilder<uint8_t> storage(size);
+  const size_t size = src.size();
+  MemBlockBuilder<uint8_t> storage(size); // Note: MemBlockBuilder takes uint64_t.
```
Comment (goaway) [inline source/common/stats/symbol_table_impl.cc:455]: In theory `MemBlockBuilder` could be updated to use `size_t`, too, and I suspect there may be other places we could switch over as well. My feeling is that that's a bit outside the scope of this change, however.

Comment (jmarantz) [reply source/common/stats/symbol_table_impl.cc:453]: agreed; outside the scope for now.

File: source/common/stats/symbol_table_impl.cc [hunk 3]
```diff
@@ -128,8 +127,7 @@ SymbolVec SymbolTableImpl::Encoding::decodeSymbols(const SymbolTable::Storage ar
 }
 
 void SymbolTableImpl::Encoding::decodeTokens(
-    const SymbolTable::Storage array, uint64_t size,
-    const std::function<void(Symbol)>& symbolTokenFn,
+    const SymbolTable::Storage array, size_t size, const std::function<void(Symbol)>& symbolTokenFn,
```
Comment (jmarantz) [inline source/common/stats/symbol_table_impl.cc:130]: not your fault, but clang-tidy is dinging you for a pre-existing naming mistake here.

s/symbolTokenFn/symbol_token_fn/ here and in the header. Sorry about that.

That will help you pass CI.

**Comment by junr03** [comment]
@ggreenway do you mind doing a last pass here?

**Pull request merged**